vision/cloud-client/detect.py

```diff
@@ -0,0 +1,389 @@
+    image = vision_client.image(source_uri=uri)
+
+    faces = image.detect_faces()
+
+    print('Faces:')
+    for face in faces:
+        print ('anger: {anger}\njoy: {joy}\nsurprise: {surprise}\n').format(
+              anger=face.emotions.anger,
+              joy=face.emotions.joy,
+              surprise=face.emotions.surprise)
+    print('')
+
+
+def detect_labels(path):
+    """Detects labels in the file."""
+    vision_client = vision.Client()
+
+    with io.open(path, 'rb') as image_file:
+        content = image_file.read()
+
+    image = vision_client.image(content=content)
+
+    labels = image.detect_labels()
+
+    print('Labels:')
+    for label in labels:
+        print(label.description)
+    print('')
+
+
+def detect_labels_cloud_storage(uri):
+    """Detects labels in the file located in Google Cloud Storage."""
+    vision_client = vision.Client()
+    image = vision_client.image(source_uri=uri)
+
+    labels = image.detect_labels()
+
+    print('Labels:')
+    for label in labels:
+        print(label.description)
+    print('')
+
+
+def detect_landmarks(path):
+    """Detects landmarks in the file."""
+    vision_client = vision.Client()
+
+    with io.open(path, 'rb') as image_file:
+        content = image_file.read()
+
+    image = vision_client.image(content=content)
+
+    landmarks = image.detect_landmarks()
+
+    print('Landmarks:')
+    for landmark in landmarks:
+        print(landmark.description)
+    print('')
+
+
+def detect_landmarks_cloud_storage(uri):
+    """Detects landmarks in the file located in Google Cloud Storage."""
+    vision_client = vision.Client()
+    image = vision_client.image(source_uri=uri)
+
+    landmarks = image.detect_landmarks()
+
+    print('Landmarks:')
+    for landmark in landmarks:
+        print(landmark.description)
+    print('')
+
+
+def detect_logos(path):
+    """Detects logos in the file."""
+    vision_client = vision.Client()
+
+    with io.open(path, 'rb') as image_file:
+        content = image_file.read()
+
+    image = vision_client.image(content=content)
+
+    logos = image.detect_logos()
+
+    print('Logos:')
+    for logo in logos:
+        print(logo.description)
+    print('')
+
+
+def detect_logos_cloud_storage(uri):
+    """Detects logos in the file located in Google Cloud Storage."""
+    vision_client = vision.Client()
+    image = vision_client.image(source_uri=uri)
+
+    logos = image.detect_logos()
+
+    print('Logos:')
+    for logo in logos:
+        print(logo.description)
+    print('')
+
+
+def detect_safe_search(path):
+    """Detects unsafe features in the file."""
+    vision_client = vision.Client()
+
+    with io.open(path, 'rb') as image_file:
+        content = image_file.read()
+
+    image = vision_client.image(content=content)
+
+    safe_searches = image.detect_safe_search()
+    print('Safe search:')
+    for safe in safe_searches:
+        print('adult: {adult}\nmedical: {medical}\nspoofed: {spoofed}\n' +
```

Comment:
If you need newlines in your print, either make it separate print statements or use heredoc.